packages/browser/src/browser.ts

```diff
@@ -0,0 +1,115 @@
+import {UAParser} from 'ua-parser-js';
+
+export interface Options {
+  userAgent: string;
+  supported?: boolean;
+  bypass?: boolean;
+}
+
+export class Browser {
+  userAgent: string;
+  supported: boolean;
+  private ua: UAParser;
+
+  get name() {
+    return this.ua.getBrowser().name || '';
+  }
+
+  get version() {
+    return this.ua.getBrowser().version || '';
+  }
+
+  get majorVersion() {
+    const {version} = this;
+
+    if (version === '') {
+      return undefined;
+    }
+
+    const majorVersion = parseInt(version.split('.')[0], 10);
+    return Number.isNaN(majorVersion) ? undefined : majorVersion;
+  }
+
+  get unknown() {
+    return this.name === '';
+  }
+
+  get isMobile() {
+    return this.ua.getDevice().type === 'mobile';
+  }
+
+  get isDesktop() {
+    return !this.isMobile;
+  }
+
+  get isShopifyMobile() {
```

Comment:
Switched `isNativeApp` to the more clearly-shopify-specific `isShopifyMobile`. Left it out of the docs because it is something only we really need. We could remove it entirely and just have our own little checks in web / other places that consume this I guess, not sure that's great though.